Modules/GamePasses.lua

```diff
@@ -12,30 +12,37 @@ local boughtGamePasses = WeakInstanceTable()
 
 local listeningPasses = {}
 
-MarketplaceService.PromptGamePassPurchaseFinished:connect(function(player, gamePassId, purchased)
+MarketplaceService.PromptGamePassPurchaseFinished:Connect(function(player, gamePassId, purchased)
 	if not purchased then return end
 	boughtGamePasses[player] = boughtGamePasses[player] or {}
 	boughtGamePasses[player][gamePassId] = true
 end)
 
 local GamePasses = {}
 
+local function userOwnsGamePassAsync(player, gamePassId)
+	local success, doesOwn = pcall(MarketplaceService.UserOwnsGamePassAsync, MarketplaceService, player.UserId, gamePassId)

```

Comment:
It's optimal, but it's unreadable, would rather this just be pcalling an anonymous function.